index.bs

```diff
@@ -895,47 +895,49 @@ spec: webidl
       </dt>
       <dd>
         A list of {{MediaDeviceInfo/deviceId}} values for the devices the user
-        has granted access to.
-      </dd>
-      <dt>
-        <a>permission result type</a>
-      </dt>
-      <dd>
-        TODO
+        has made a non-default decision on access to.
       </dd>
       <dt>
         <a>permission query algorithm</a>
       </dt>
       <dd>
-        TODO
+        The permission query algorithm runs the following steps:
+        <ol class="algorithm">
+        <li>
+        If |permissionDesc|.deviceId exists in the <a>extra permission data</a>,
+        set <code><var>status</var>.state</code>
+        to |permissionDesc|'s <a>permission state</a> and terminate these
+        steps.
+        </li>
+        <li>Set <var>global</var> to the {{PermissionDesc}} for the descriptor
+        that results from removing the "deviceId" value.
+        </li>
+        <li>
+        Set <code><var>status</var>.state</code> to <var>global</var>'s
+        <a>permission state</a>.
+        </li>
       </dd>
       <dt>
         <a>permission request algorithm</a>
       </dt>
       <dd>
-        TODO
+        Run the <a>boolean permission request algorithm</a>
```

Comment:
I suspect `request()` isn't very useful when it can only request a device ID the site already knows about. Folks are likely to want to pass the same arguments they can pass to `getUserMedia()`, and to get a stream back.